python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py

```diff
@@ -327,5 +321,26 @@ def _task_definition(self, family, metadata, image, container_context):
             include_sidecars=self.include_sidecars,
         )
 
+    def _reuse_task_definition(self, task_definition, metadata, image, secrets):
+        container_definitions_match = False
+        task_definitions_match = False
+
+        container_definitions = task_definition.get("containerDefinitions", [{}])
+        # Only check for diffs to the primary container. This ignores changes to sidecars.
```

Comment:
Added this comment because I noticed this behavior - a more robust check would compare each container definition to a container definition with the same name. Probably something we should do eventually, but unrelated to this change.